readthedocs/rtd_tests/tests/test_sync_versions.py

```diff
@@ -152,6 +153,498 @@ def test_new_tag_update_inactive(self):
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        # The version 8 should be the new stable.
+        # The stable isn't stuck with the previous commit
+        version8 = self.pip.versions.get(slug='0.8.3')
+        current_stable = self.pip.get_stable_version()
+        self.assertEqual(
+            version8.identifier,
+            current_stable.identifier,
+        )
+        self.assertNotEqual(
+            'origin/stable',
+            current_stable.identifier,
+        )
+        self.assertTrue(current_stable.machine)
+
+    @pytest.mark.xfail(strict=True)
+    def test_normal_behavior_for_latest_after_deleting_user_defined_tag(self):
+        """
+        The user creates a tag named ``latest`` on an existing repo,
+        when syncing the versions, the RTD's ``latest`` is lost
+        and doesn't update automatically anymore, when the tag is deleted
+        on the user repository, the RTD's ``latest`` is back.
+        """
+        version_post_data = {
+            'branches': [
+                {
+                    'identifier': 'origin/master',
+                    'verbose_name': 'master',
+                },
+            ],
+            'tags': [
+                # User new stable
+                {
+                    'identifier': '1abc2def3',
+                    'verbose_name': 'latest',
+                },
+            ],
+        }
+
+        resp = self.client.post(
+            '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
+            data=json.dumps(version_post_data),
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        # The tag is the new latest
+        version_latest = self.pip.versions.get(slug='latest')
+        self.assertEqual(
+            '1abc2def3',
+            version_latest.identifier
+        )
+
+        # Deleting the tag should return the RTD's latest
+        version_post_data = {
+            'branches': [
+                {
+                    'identifier': 'origin/master',
+                    'verbose_name': 'master',
+                },
+            ],
+        }
+
+        resp = self.client.post(
+            '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
+            data=json.dumps(version_post_data),
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        # The latest isn't stuck with the previous commit
+        version_latest = self.pip.versions.get(slug='latest')
+        self.assertEqual(
+            'origin/master',
+            version_latest.identifier,
+        )
+        self.assertNotEqual(
+            '1abc2def3',
+            version_latest.identifier,
+        )
+        self.assertTrue(version_latest.machine)
+
+    @pytest.mark.xfail(strict=True)
+    def test_normal_behavior_for_latest_after_deleting_user_defined_branch(self):
```

Comment:
Are we blocked on a decission to make before making this test to pass?

Reply:
Nope, just trying to understand all cases for stable first